backend/tests/test_settlement_calculation.py

```
@@ -0,0 +1,950 @@
+The key insight from debugging the "Ins valsura internship" group:
+- For EVERY transaction (expense or payment), each user has:
+  - paidShare: amount they contributed
+  - owedShare: amount they should pay
+  - netEffect = paidShare - owedShare
+    - Positive = they are owed money (creditor)
+    - Negative = they owe money (debtor)
+- Settlements are created by matching debtors to creditors
+"""
+
+from collections import defaultdict
+from typing import Any, Dict, List, Tuple
```

Ruff (0.14.11)

`typing.Dict` is deprecated, use `dict` instead

(UP035)